test/client/spec/components/victory-candlestick/victory-candlestick.spec.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * Client tests
+ */
+/*eslint-disable max-nested-callbacks */
+/*eslint-disable no-console */
+/*eslint-disable no-undef */
+/* global sinon */
+/* eslint no-unused-expressions: 0 */
+import React from "react";
+import { shallow, mount } from "enzyme";
+import { range, omit } from "lodash";
+import VictoryCandlestick from "src/components/victory-candlestick/victory-candlestick";
+import Candle from "src/components/victory-candlestick/candle";
+import { VictoryLabel } from "victory-core";
+
+class MyCandle extends React.Component {
+
+  render() { }
+}
+
+const dataSet = [{x: 5, open: 10, close: 20, high: 25, low: 5},
+{x: 1, open: 80, close: 40, high: 120, low: 10, label: "1"}];
+
+describe("components/victory-candlestick", () => {
+  describe("default component rendering", () => {
+    it("renders an svg with the correct width and height", () => {
+      const wrapper = mount(
+        <VictoryCandlestick data={dataSet}/>
+      );
+      const svg = wrapper.find("svg");
+      expect(svg.prop("style").width).to.equal("100%");
+      expect(svg.prop("style").height).to.equal("auto");
+    });
+
+    it("renders an svg with the correct viewBox", () => {
+      const wrapper = mount(
+        <VictoryCandlestick data={dataSet}/>
+      );
+      const svg = wrapper.find("svg");
+      const viewBoxValue =
+        `0 0 ${VictoryCandlestick.defaultProps.width} ${VictoryCandlestick.defaultProps.height}`;
+      expect(svg.prop("viewBox")).to.equal(viewBoxValue);
+    });
+
+    it("renders 51 points", () => {
+      const wrapper = shallow(
+        <VictoryCandlestick/>
+      );
+      const points = wrapper.find(Candle);
+      expect(points.length).to.equal(51);
+    });
+  });
+
+  describe("rendering data", () => {
+    it("renders injected points for {x, y} shaped data (default)", () => {
+      const data = range(10).map((i) => ({x: i, y: i}));
+      const wrapper = shallow(
+        <VictoryCandlestick data={data} dataComponent={<MyCandle />} />
+      );
+
+      const points = wrapper.find(MyCandle);
+      expect(points.length).to.equal(10);
+    });
+
+    it("renders points for {x, y} shaped data (default)", () => {
+      const data = range(10).map((i) => ({x: i, y: i}));
+      const wrapper = shallow(
+        <VictoryCandlestick data={data}/>
+      );
+      const points = wrapper.find(Candle);
+      expect(points.length).to.equal(10);
+    });
+
+    it("renders points for array-shaped data", () => {
+      const data = range(20).map((i) => [i, i]);
+      const wrapper = shallow(
+        <VictoryCandlestick data={data} x={0} y={1}/>
+      );
+      const points = wrapper.find(Candle);
+      expect(points.length).to.equal(20);
+    });
+
+    it("renders points for deeply-nested data", () => {
+      const data = range(40).map((i) => ({a: {b: [{x: i, y: i}]}}));
+      const wrapper = shallow(
+        <VictoryCandlestick data={data} x="a.b[0].x" y="a.b[0].y"/>
```

Comment:
I would expect this spec to fail since you don't have high / low / open / close defined in your data or defined as accessors